textile/fabric_printing/report/pretreatment_production_register/pretreatment_production_register.py

```diff
@@ -0,0 +1,336 @@
+# Copyright (c) 2023, ParaLogic and contributors
+# For license information, please see license.txt
+
+import frappe
+from frappe import _, scrub, unscrub
+from frappe.utils import cint, cstr, flt, getdate, add_days
+from frappe.desk.query_report import group_report_data
+from erpnext.setup.doctype.uom_conversion_factor.uom_conversion_factor import get_uom_conv_factor
+
+
+def execute(filters=None):
+	return PretreatmentProductionRegister(filters).run()
+
+
+class PretreatmentProductionRegister:
+	def __init__(self, filters=None):
+		self.filters = frappe._dict(filters or {})
+		self.filters.from_date = getdate(filters.from_date)
+		self.filters.to_date = getdate(filters.to_date)
+
+		if self.filters.from_date > self.filters.to_date:
+			frappe.throw(_("Date Range is incorrect"))
+
+		self.show_item_name = frappe.defaults.get_global_default('item_naming_by') != "Item Name"
+		self.show_customer_name = frappe.defaults.get_global_default('cust_master_name') == "Naming Series"
+
+	def run(self):
+		self.get_data()
+		self.prepare_data()
+		self.get_chart_data()
+		self.data = self.get_grouped_data()
+		self.get_columns()
+
+		return self.columns, self.data, None, self.chart_data
+
+	def get_data(self):
+		conditions = self.get_conditions()
+
+		self.data = frappe.db.sql("""
+			SELECT se.name as stock_entry, se.posting_date, se.posting_time,
+				timestamp(se.posting_date, se.posting_time) as posting_dt,
+				se.work_order, se.fg_completed_qty as qty,
+				wo.pretreatment_order, wo.stock_uom as uom,
+				wo.customer, wo.customer_name,
+				wo.production_item as ready_fabric, wo.item_name as ready_fabric_name,
+				wo.fabric_item as greige_fabric, wo.fabric_item_name as greige_fabric_name,
+				item.net_weight_per_unit, item.weight_uom
+			FROM `tabStock Entry` se
+			INNER JOIN `tabWork Order` wo
+				ON wo.name = se.work_order
+			LEFT JOIN `tabItem` item
+				ON item.name = wo.fabric_item
+			WHERE se.docstatus = 1
+				AND se.posting_date between %(from_date)s AND %(to_date)s
+				AND se.purpose = 'Manufacture'
+				AND ifnull(wo.pretreatment_order, '') != ''
+				{conditions}
+			ORDER BY se.posting_date, se.posting_time
+		""".format(conditions=conditions), self.filters, as_dict=1)
+
+		ready_fabrics = list(set([d.ready_fabric for d in self.data]))
+		self.square_meter_conversion = {}
+
+		if ready_fabrics:
+			self.square_meter_conversion = dict(frappe.db.sql("""
+				select parent, conversion_factor
+				from `tabUOM Conversion Detail`
+				where parenttype = 'Item' and parent in %s and uom = 'Square Meter'
+			""", [ready_fabrics]))
```

Comment:
use raw material fabric's (greige fabric) square meters
update in print production register as well